worker.go

```diff
@@ -95,14 +95,14 @@ func (worker *worker) startNewWorkerThread() {
 	backingOffLock := sync.RWMutex{}
 
 	for {
-
 		// if the worker can stay up longer than backoff*2, it is probably an application error
 		upFunc := sync.Once{}
 		go func() {
 			backingOffLock.RLock()
 			wait := backoff * 2
 			backingOffLock.RUnlock()
 			time.Sleep(wait)
+			worker.ready <- struct{}{}
```

Comment:
I'm not sure if this is the correct approach. What if the worker script takes longer than 200ms to start and then fails? Additionally, this will unnecessarily delay startup in most cases.
This line should probably be moved somewhere to `metrics.ReadyWorker`, right after `frankenphp_handle_request()` was called.

Reply:
Currently it will definitely wait for all threads to be ready via the `workersReadyWG`, which only reaches 0 after each thread has called `frankenphp_handle_request()`. With this implementation we could reach a 'ready' state even if the worker script doesn't contain `frankenphp_handle_request()`
I guess it depends on when you would consider the worker thread to be 'ready', but why go with 'reasonable confidence' when you can have absolute confidence instead? (and be faster in most cases)

Reply:
> Currently it will definitely wait for all threads to be ready via the workersReadyWG, which only reaches 0 after each thread has called frankenphp_handle_request(). 

I honestly don't remember what I ran into last night that specifically triggered this rabbit hole, other than turning on the watcher caused all kinds of problems with the metrics being output. TL;DR: I was just trying to simplify things to understand _why_ it is causing issues, then running into other issues as I went, yak shaving my way through it all.

I ran into a number of minor issues that sometimes made it past the atomics, like restarting during shutdown. Atomics are terrible unless you are using CAS, and you are not guaranteed to get the latest value ... so I deleted them. From there, it was just figuring out what was going wrong with the `WorkersReadyWG` because as mentioned, the current architecture doesn't really make sense for them, but we don't want to run tests until workers are started.

So, most of the refactoring is just in getting rid of atomics. This then uncovered new failing tests 100% of the time instead of randomly, particularly when restarting workers... and so on.

After all this, I still don't know why watchers specifically caused my issues 😆

Reply:
Yeah I originally intended my refactor PR to just be a small step towards more control over PHP threads. But then I realized it could potentially also fix Fibers and it kinda ended up doing more than I wanted it to. 
Problem with these big PRs is that they kind of block other PRs (like this one).

Reply:
Maybe we should take smaller steps in our PR. Unfortunately, I never did figure out what was broken here, short of refactoring the entire worker stuff. :(

It might be worth working on an actual design in a Discussion? That way we'd be able to work on different parts independently and break it down into smaller PRs. Maybe we could take a 'strangler fig' approach and reimplement it separate from the current codebase. So, we could just use build tags to switch between worker implementations (at least while we are working on it), and people could test it more easily.

Reply:
https://github.com/dunglas/frankenphp/discussions/1175